plugin.py

```diff
@@ -99,3 +115,74 @@ def find_package_dependency_dirs(self, py_ver: Tuple[int, int] = (3, 3)) -> List
         dep_dirs.insert(0, os.path.join(self.package_storage(), "resources", "typings", "sublime_text"))
 
         return [path for path in dep_dirs if os.path.isdir(path)]
+
+    @classmethod
+    def resolve_python_path_from_venv(
+        cls, settings: DottedDict, workspace_folders: List[WorkspaceFolder]
+    ) -> Optional[str]:
+        """
+        Resolves the python binary path depending on environment variables and files in the workspace.
+
+        See https://github.com/fannheyward/coc-pyright/blob/d58a468b1d7479a1b56906e386f44b997181e307/src/configSettings.ts#L47.  # noqa: E501
+        """
+
+        def binary_from_python_path(path: str) -> Optional[str]:
+            if sublime.platform() == "windows":
+                binary_path = os.path.join(path, "Scripts", "python.exe")
+            else:
+                binary_path = os.path.join(path, "bin", "python")
+
+            return binary_path if os.path.isfile(binary_path) else None
+
+        pythonPath = settings.get("python.pythonPath")
+        if pythonPath:
+            return pythonPath
+
+        virtual_env = os.getenv("VIRTUAL_ENV")
+        if virtual_env is not None:
+            return binary_from_python_path(virtual_env)
+
+        conda_prefix = os.getenv("CONDA_PREFIX")
```

Comment:
How much sense does it make to read environment variables that are ST-wide rather than specific to the directory of the project? Would it ever make sense that someone would have those set per-system?

Reply:
Good point. Is there a possibility to query the environment that is set when `subl .` is executed?

Reply:
Environment is inherit on start of ST and it won't change on running `subl .` if it's already open. Using environment inherited at start doesn't seem robust as then the behavior will differ depending on how ST was started.

I suppose it might be better to start a subprocess in the workspace folder and read environment from that but I'm not clean on how would `VIRTUAL_ENV` and `CONDA_PREFIX` be set in a typical scenario. Is it on directory change or would require manual execution of some custom script? If the latter then I don't see it being feasible to support that.

Reply:
Are you sure that the subprocess won't just inherit the sublime environment and output the same as `os.getenv`?

Reply:
I think @rchl means if the user has to do something like `CONDA_PREFIX=FOO subl .` then after ST starts, `CONDA_PREFIX` is like a fixed value even if the user switches project without restarting ST.

Reply:
> Are you sure that the subprocess won't just inherit the sublime environment and output the same as `os.getenv`?

It does unless `env` subprocess argument is overridden.

Now the question is how `CONDA_PREFIX` and `VIRTUAL_ENV` are typically set. If for example through some hook in default shell that sets those on changing directories then we could potentially get those from the subprocess call. But I feel like it's more manual in nature so think that subprocess call might not help much.

Reply:
Yes I think it's best to just skip reading ENV variables.